snuba/snuba_migrations/profile_chunks/0001_create_profile_chunks_table.py

```diff
@@ -0,0 +1,85 @@
+from typing import List, Sequence
+
+from snuba.clickhouse.columns import UUID, Column, DateTime64, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.operations import OperationTarget, SqlOperation
+
+storage_set = StorageSetKey.PROFILE_CHUNKS
+table_prefix = "profile_chunks"
+local_table_name = f"{table_prefix}_local"
+dist_table_name = f"{table_prefix}_dist"
+
+UNKNOWN_SPAN_STATUS = 2
+
+columns: List[Column[Modifiers]] = [
+    Column("project_id", UInt(64)),
+    Column("profiler_id", UUID()),
+    Column("chunk_id", UUID()),
+    Column(
+        "start_timestamp",
+        DateTime64(
+            precision=6,
+            modifiers=Modifiers(codecs=["DoubleDelta"]),
+        ),
+    ),
+    Column(
+        "end_timestamp",
+        DateTime64(
+            precision=6,
+            modifiers=Modifiers(codecs=["DoubleDelta"]),
+        ),
+    ),
+    Column("retention_days", UInt(16)),
+    Column("partition", UInt(16)),
+    Column("offset", UInt(64)),
+    Column("deleted", UInt(8)),
+]
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+
+    def forwards_ops(self) -> Sequence[SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=storage_set,
+                table_name=local_table_name,
+                columns=columns,
+                engine=table_engines.ReplacingMergeTree(
+                    order_by="(project_id, cityHash64(profiler_id), start_timestamp, end_timestamp, cityHash64(chunk_id))",
```

Comment:
Will we have a processor that maps `profiler_id = ...` conditions into `cityHash64(profiler_id) = cityHash64(...)`? Without it, I believe queries won't fully take advantage of the order by for max performance.

Reply:
What is the purpose of hashing the profiler id?

Reply:
I'm also wondering if the `end_timestamp` is necessary? I can't imagine profiles being so far apart that you need both.

Reply:
The purpose of hashing the `profiler_id` is for the sampling key (it needs to be an integer) but @Zylphrex and I already agreed it `chunk_id` might be a better sampling key anyway. I'll update to that and remove the hashing of `profiler_id`.

`end_timestamp` is not necessarily needed, I suppose I wanted to maximize the use of the index but maybe that's trying to do too much.

Reply:
I would leave end_timestamp out of it. The order_by key is used to whittle down how many parts you scan and it's likely you're not going to cut down on any parts. You'll just make the sorting key bigger. Tbh I don't know what the negative impact of that will be but I don't see a positive one either. 

If you're interested in an explanation of query processing:
https://docs.google.com/presentation/d/1-pwlFWx05sPqkv6WYn4TQZFWfLzw1wAoE8eykq_aL2s/edit#slide=id.g229f20e6452_0_892